lib/SILGen/SILGenApply.cpp

```diff
@@ -1117,12 +1117,19 @@ class SILGenApply : public Lowering::ExprVisitor<SILGenApply> {
 
     // Otherwise, we have a statically-dispatched call.
     auto constant = SILDeclRef(e->getDecl());
-    if (e->getDecl()->getAttrs().hasAttribute<DistributedActorAttr>()) {
+    if (afd && afd->isDistributed()) {
       // we're calling a distributed function, only in cross-actor situations
       // does this actually need to call the thunk.
-      if (!selfApply) {
+      fprintf(stderr, "[%s:%d] (%s) EXPR, e->isImplicitlyThrows() == %d:\n", __FILE__, __LINE__, __FUNCTION__,
+              e->isImplicitlyThrows()x);
+      e->dump();
+
+      // TODO: this is a DeclRefEpr but we need the ApplyExpr which had the implicit throwing set on it...
+      //       how to get it? or do we need to set on the ref instead?
+      if (e->isImplicitlyThrows()) {
```

Comment:
if this was the Expr that we tryMarkImplicitly-throw-ed then it'd work... but it's  not, it is: 

```
[/Users/ktoso/code/swift-project/swift/lib/SILGen/SILGenApply.cpp:1123] (visitDeclRefExpr) EXPR, e->isImplicitlyThrows() == 0:
(declref_expr type='(isolated SomeSpecificDistributedActor) -> () throws -> String' location=/Users/ktoso/code/swift-project/swift/test/Distributed/Runtime/distributed_actor_remote_functions.swift:162:29 range=[/Users/ktoso/code/swift-project/swift/test/Distributed/Runtime/distributed_actor_remote_functions.swift:162:29 - line:162:29] decl=main.(file).SomeSpecificDistributedActor.helloThrowsImplBoom()@/Users/ktoso/code/swift-project/swift/test/Distributed/Runtime/distributed_actor_remote_functions.swift:46:20 function_ref=double)
```